api/envoy/data/cluster/v3/outlier_detection_event.proto

```diff
@@ -144,3 +150,11 @@ message OutlierEjectFailurePercentage {
   // Host's success rate at the time of the ejection event on a 0-100 range.
   uint32 host_success_rate = 1 [(validate.rules).uint32 = {lte: 100}];
 }
+
+message OutlierEjectExtension {
+  // Name of extension monitor which caused a host ejection.
+  string monitor_name = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
What name format is expected here? The full extension name like `envoy.outlier_detection_monitors.consecutive_errors` or a shorter name like `consecutive_errors`? The former one is clear but seems is verbose for recording the event.

Reply:
The monitor_name will come from configuration: https://github.com/envoyproxy/envoy/blob/6ca3260c12ee7609a372e79910111df455d22855/api/envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.proto#L26. So, if you configure the consecutive monitor with name "my_5xx_monitor", that name will be used when event is logged.

Reply:
Hah, I see. But wait, what's the difference between the `name` here and the `TypedExtensionConfig.name`?

IMO, we shouldn't add additional `name` here and should use the `TypedExtensionConfig.name` directly. Note, now the name in the `TypedExtensionConfig`  is used to represent the unique extension instance, not to represent the extension type.

Reply:
Good point. Let me check if I can use extension's name and get the similar results. Thanks!

Reply:
Done. Removed name from monitor's proto and name attached to typed extension is used. It actually fits better the code this way. Thanks for catching it!